napari/layers/_labels_layer/model.py

```diff
@@ -161,7 +341,8 @@ def _slice_image(self, indices, image=None):
             except TypeError:
                 pass
 
-        return image[tuple(indices)]
+        slice_indices = tuple(indices)
+        return image[slice_indices], slice_indices
```

Comment:
I'm worried that this is making the layer diverge too much from the Image layer. In general, it's not true that you always want the slice indices returned. How about putting the `slice_indices` logic in a separate method, and then doing:

```python
def _slice_image(self, indices, image=None):
    image = image or self._image
    slice_indices = self._get_indices(indices, image=image)
    return np.ndarray(image[slice_indices])  # add np.array cast to allow lazy arrays, as in Image
```

Reply:
Although, note that this will break when writing out to the big array — that's something that we will have to solve separately I think.

Reply:
i have changed this. we might need to revisit when look at the big array stuff. not sure i'm quite ready for that in this PR